dbt/adapters/default/impl.py

```diff
@@ -324,10 +361,36 @@ def expand_target_column_types(self,
     ###
     # RELATIONS
     ###
-    def list_relations(self, schema, model_name=None):
+    def _is_cached(self, schema, model_name=None, debug_on_missing=True):
+        """Check if the schema is cached, and by default logs if it is not."""
+        if dbt.flags.USE_CACHE is False:
+            return False
+        elif schema not in self.cache:
+            if debug_on_missing:
+                logger.debug(
+                    'On "{}": cache miss for schema "{}", this is inefficient'
+                    .format(model_name or '<None>', schema)
+                )
+            return False
+        else:
+            return True
```

Comment:
Is this method only intended to return if a `schema` is cached? I imagined that the `else` branch would check if `model_name` is present in the cache from looking at the signature and how it's used

Reply:
The cache only tracks 'in' status on a schema level, it's impossible to know if an entry is unknown to the cache or actually does not exist. The model_name bit is just for logging.

I'll rename the method to try to communicate that better.